### PR TITLE
h5py 2.5.0 built using the foss-2016a toolchain

### PR DESCRIPTION
**Pull request by FokkeDijkstra**
h5py 2.5.0 built using the foss-2016a toolchain. Both for python 2.7.11 and 3.5.1. The latter requires a patch in order for it to built correctly.

### CODE REVIEW

File: easybuild/easyconfigs/h/h5py/h5py-2.5.0-foss-2016a-Python-3.5.1-HDF5-1.8.16.eb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+source_urls = [PYPI_SOURCE]
+sources = [SOURCE_TAR_GZ]
+
+hdf5ver = '1.8.16'
+versionsuffix = '-Python-%%(pyver)s-HDF5-%s' % hdf5ver
+
+# to really use mpi enabled hdf5 we now seem to need a configure step
+prebuildopts = ' python setup.py configure --mpi --hdf5=$EBROOTHDF5 && '
+
+patches = ['h5py-2.5.0-foss-2016a-Python-3.5.1-HDF5-1.8.16.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/h/h5py/h5py-2.5.0-foss-2016a-Python-3.5.1-HDF5-1.8.16.eb]: don't hardcode the `h5py` version in the patch file name, please use `%(version)s`

also: move this up below `sources`

File: easybuild/easyconfigs/h/h5py/h5py-2.5.0-foss-2016a-Python-3.5.1-HDF5-1.8.16.eb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+toolchainopts = {'usempi': True}
+
+source_urls = [PYPI_SOURCE]
+sources = [SOURCE_TAR_GZ]
+
+hdf5ver = '1.8.16'
+versionsuffix = '-Python-%%(pyver)s-HDF5-%s' % hdf5ver
+
+# to really use mpi enabled hdf5 we now seem to need a configure step
+prebuildopts = ' python setup.py configure --mpi --hdf5=$EBROOTHDF5 && '
```
Comment (boegel) [inline easybuild/easyconfigs/h/h5py/h5py-2.5.0-foss-2016a-Python-3.5.1-HDF5-1.8.16.eb]: ideally this goes below `dependencies`

**Comment by boegel** [comment]
@FokkeDijkstra tests are failing because of missing easyconfig file:

```
EasyBuildError: 'Irresolvable dependencies encountered: pkgconfig/1.1.0-foss-2016a-Python-3.5.1'
```

**Comment by FokkeDijkstra** [comment]
pkgconfig for Python 3.5.1 has been added in  #3297

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2433.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/393dc09d0e12cfc25a07f3f21ed4e93c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2019.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/81c1af725caa670385333fc994c462ad for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @FokkeDijkstra!

**Pull request merged**